File: Longest Common Prefix/Remove_All_Occurrences_of_a_Substring.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        mpp = defaultdict(int)
        n = len(strs)

        for s in strs:
            for j in range(len(s)):
                temp = s[:j + 1]
                mpp[temp] += 1

        ans = ""
        for key, value in mpp.items():
            if value == n and len(key) > len(ans):
                ans = key

        return ans

File: Longest Common Prefix/Remove_All_Occurrences_of_a_Substring.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        unordered_map<string,int> mpp;
        for(auto i: strs) {
            for(int j = 0; j<i.size(); j++) {
                string temp = i.substr(0,j+1);
                mpp[temp]++;
            }
        }
        int n = strs.size();
        string ans = "";
        for(auto i: mpp) {
            if(i.second == n) {
                if(ans.size() < i.first.size()) {
                    ans = i.first;
                }
            }
        }
        return ans;
    }
};